src/context/AuthContext.js

```diff
@@ -1,34 +1,109 @@
-import React, { createContext, useState } from 'react';
+import React, {createContext, useEffect, useState} from 'react';
 import { useNavigate } from 'react-router-dom';
+import jwt_decode from "jwt-decode";
+import {checkTokenValidity} from "../helper/checkTokenValidity";
+import axios from "axios";
 
 export const AuthContext = createContext({});
 
 function AuthContextProvider({ children }) {
-  const [isAuth, toggleIsAuth] = useState(false);
+  const [auth, toggleAuth] = useState({
+    isAuth: false,
+    user: null,
+    secret: null,
+    status: "pending"
+  });
+  console.log(auth.isAuth)
+
   const navigate = useNavigate();
 
-  function login() {
-    console.log('Gebruiker is ingelogd!');
-    toggleIsAuth(true);
-    navigate('/profile');
+  useEffect(() => {
+    const storedToken = localStorage.getItem('token');
+
+    if (storedToken && checkTokenValidity(storedToken)) {
```

Comment:
Ik vind de `checkTokenValidity` meer binnen de `login` functie thuishoren 😉

Reply:
Op deze manier is het in de les geleerd/voorgedaan... 🤔

Reply:
Het is ook niet zo zeer fout, maar als je op meer dan één plek de login functie gaat aanroepen, maar altijd daarvoor de `checkTokenValidity` functie moet aanroepen, dan is het beter om deze te combineren. Dat scheelt weer wat code 😉